source/python/idetect/nlp_models/relevance.py

```diff
@@ -1,204 +1,235 @@
-import gensim
 import numpy as np
 import pandas as pd
-from nltk.stem import PorterStemmer
-from nltk.tokenize import WordPunctTokenizer
+import re
+import string
 from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.pipeline import Pipeline, FeatureUnion
+from sklearn.svm import LinearSVC
+from spacy.tokens.token import Token
 
 from idetect.model import Relevance
-from idetect.nlp_models.base_model import DownloadableModel
+from idetect.nlp_models.base_model import DownloadableModel, CustomSklLsiModel
+from idetect.fact_extractor import nlp
+from idetect.geotagger import strip_accents, compare_strings, strip_words, common_names, LocationType, subdivision_country_code, match_country_name, city_subdivision_country
 
 
 class RelevanceModel(DownloadableModel):
-    def __init__(self, model_path='relevance.pkl',
-            model_url='https://s3-us-west-2.amazonaws.com/idmc-idetect/relevance_models/relevance.pkl'):
+    def __init__(self, model_path='/home/idetect/python/idetect/nlp_models/relevance_classifier_svm_10052017.pkl',
+            model_url='https://s3-us-west-2.amazonaws.com/idmc-idetect/relevance_models/relevance_classifier_svm_10052017.pkl'):
         self.model = self.load_model(model_path, model_url)
 
     def predict(self, text):
         try:
-            relevance = self.model.transform(pd.Series(text))[0]
-            if relevance == 'yes':
+            relevance = self.model.predict(pd.Series(text))[0]
+            if relevance == 1:
                 return Relevance.DISPLACEMENT
-            else:
+            elif relevance == 0:
                 return Relevance.NOT_DISPLACEMENT
         except ValueError:
             # error can occur if empty text is passed to model
             raise
 
 
-class Tokenizer(TransformerMixin):
-    def __init__(self, stop_words=None):
-        self.stop_words = stop_words
-        self.tokenizer = WordPunctTokenizer()
-
-    def prepare_tokens(self, text, stop_words):
-        tokens = self.tokenizer.tokenize(text)
-        tokens = [t for t in tokens if len(t) > 2]
-        tokens = [t.lower() for t in tokens]
-        tokens = [t for t in tokens if t not in stop_words]
-        tokens = [t for t in tokens if not t.isdigit()]
+class CleaningProcessor(BaseEstimator, TransformerMixin):
+    """Transformer that turns documents in string form
+    into token lists, with various processing steps applied.
+
+    Parameters
+    ----------
+    pos_tags : bool, required
+        Whether to tag words with their part of speech labels.
+    lemmatize : bool, required
+        Whether to lemmatize tokens.
+    stop_words : book, required
+        Whether to remove stop words.
+    """
+
+    def cleanup(self, text):
+        """
```

Comment:
You can import `cleanup` from `idetect.model` and that way we have a single function definition